airflow/utils/log/es_task_handler.py

```diff
@@ -67,19 +72,37 @@ def __init__(self, base_log_folder, filename_template,
 
         self.mark_end_on_close = True
         self.end_of_log_mark = end_of_log_mark
+        self.write_stdout = write_stdout
+        self.json_format = json_format
+        self.json_fields = [label.strip() for label in json_fields.split(",")]
+
+        self.handler = None
 
     def _render_log_id(self, ti, try_number):
         if self.log_id_jinja_template:
             jinja_context = ti.get_template_context()
             jinja_context['try_number'] = try_number
             return self.log_id_jinja_template.render(**jinja_context)
 
+        if self.json_format:
+            execution_date = self._clean_execution_date(ti.execution_date)
+        else:
+            execution_date = ti.execution_date.isoformat()
         return self.log_id_template.format(dag_id=ti.dag_id,
                                            task_id=ti.task_id,
-                                           execution_date=ti
-                                           .execution_date.isoformat(),
+                                           execution_date=execution_date,
                                            try_number=try_number)
 
+    @staticmethod
+    def _clean_execution_date(execution_date):
+        """
+        Clean up an execution date so that it is safe to query in elasticsearch
+        by removing reserved characters.
+        # https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html#_reserved_characters
+        :param execution_date: execution date of the dag run.
+        """
+        return re.sub(r"[\+\-\:\.]", "", execution_date.isoformat())
```

Comment:
Might it be better to use `execution_date.strftime(something)` instead?

Doesn't ES support datetimes in ISO8601 format thought?

Reply:
@ashb  what do you think if it should be:
```
In [5]: dt.today().strftime("%Y_%m_%dT%H_%I_%S_%f")
Out[5]: '2019_04_18T15_03_06_899836'
```
cc @schnie

Reply:
or
```
In [13]: re.sub(r"[\+\-\:\.]", "", dt_str)
Out[13]: '20190418T155643400365'

In [14]: re.sub(r"[\+\-\:\.]", "_", dt_str)
Out[14]: '2019_04_18T15_56_43_400365'
```

Reply:
Where does this show up btw? Using a strftime feels better rather than generating it in one format and then stripping some characters out.

Reply:
@ashb these get output as part of the json output of the log message. We then use those fields in our fluentd config (or any other log shipping tool), to construct the `log_id` here: https://github.com/astronomer/helm.astronomer.io/blob/master/charts/fluentd/templates/fluentd-configmap.yaml#L160. The es log handler then runs its queries using the `log_id`.

Reply:
Oh this forms part of the log ID, not the timestamp of the individual message. Gotcha.

Reply:
Why not use pendulum for datetime formatting? This is prone to errors

Reply:
@bolkedebruin  can you suggest something similar to `dt.today().strftime("%Y_%m_%dT%H_%I_%S_%f")` from 
(pendulum docs)[https://pendulum.eustace.io/docs/#string-formatting]?